commons-numbers-complex-arrays/src/main/java/org/apache/commons/numbers/complex/arrays/ComplexList.java

```diff
@@ -0,0 +1,357 @@
+        double[] realAndImgData = new double[c.size() * 2];
+        int i = 0;
+        for (final Complex val : c) {
+            final int i2 = i << 1;
+            realAndImgData[i2] = val.getReal();
+            realAndImgData[i2 + 1] = val.getImaginary();
+            i++;
+        }
+        System.arraycopy(realAndImgData, 0, realAndImagParts, size * 2, realAndImgData.length);
+        size += numNew;
+        return numNew != 0;
+    }
+
+    /**
+     * Inserts all the elements in the specified collection into this list, starting at the specified position.
+     * Shifts the element currently at that position (if any) and any subsequent elements to the right (increases their indices).
+     * The new elements will appear in the list in the order that they are returned by the specified collection's iterator.
+     *
+     * @param index – Index at which to insert the first element from the specified collection.
+     * @param c – Collection containing elements to be added to this list.
+     * @return true if this list changed as a result of the call.
+     * @throws IndexOutOfBoundsException – if index isn't within the range of list.
+     * @throws NullPointerException – if the specified collection is null.
+     */
+    @Override
+    public boolean addAll(int index, Collection<? extends Complex> c) {
+        rangeCheckForInsert(index);
+        final int numNew = c.size();
+        final int numNew2 = numNew << 1;
+        expand(numNew * 2);
+        final double[] realAndImgData = new double[c.size() * 2];
+        int i = 0;
+        for (final Complex val : c) {
+            final int i2 = i << 1;
+            realAndImgData[i2] = val.getReal();
+            realAndImgData[i2 + 1] = val.getImaginary();
+            i++;
+        }
+        final int numMoved = (size - index) * 2;
+        final int index2 = index << 1;
+        System.arraycopy(realAndImagParts, index2, realAndImagParts, index2 + numNew2, numMoved);
+        System.arraycopy(realAndImgData, 0, realAndImagParts, index2, realAndImgData.length);
+        size += numNew;
+        return numNew != 0;
+    }
+
+    /**
+     * Removes the element at the specified position in this list.
+     * Shifts any subsequent elements to the left (subtracts one from their indices).
+     *
+     * @param index – Index of the element to be removed.
+     * @return the element that was removed from the list.
+     * @throws IndexOutOfBoundsException – if index isn't within the range of list.
+     */
+    @Override
+    public Complex remove(int index) {
+        rangeCheck(index);
+        final int i = index << 1;
+        final int s = size << 1;
+        final Complex oldValue = Complex.ofCartesian(realAndImagParts[i], realAndImagParts[i + 1]);
+        final int numMoved = s - i - 2;
+        if (numMoved > 0) {
+            System.arraycopy(realAndImagParts, i + 2, realAndImagParts, i, numMoved);
+        }
+        size--;
+        realAndImagParts[s] = 0;
```

Comment:
This has no purpose as this part of the array should never be accessed as it is out-of-range.